aspnetcore/fundamentals/configuration.md

```diff
@@ -5,162 +5,981 @@ ms.author: riande
+
+<!-- literal_block {"xml:space": "preserve", "language": "none", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "source": "/Users/shirhatti/src/Docs/aspnet/fundamentals/configuration/sample/src/ObjectGraph/Program.cs", "highlight_args": {"linenostart": 1}, "names": []} -->
+
+````none
+
+   using Microsoft.Extensions.Configuration;
+   using System;
+   using System.IO;
+
+   // Add these NuGet packages:
+   // "Microsoft.Extensions.Configuration.Binder"
+   // "Microsoft.Extensions.Configuration.FileExtensions"
+   // "Microsoft.Extensions.Configuration.Json": 
+   public class Program
+   {
+       public static void Main(string[] args = null)
+       {
+           var builder = new ConfigurationBuilder()
+                      .SetBasePath(Directory.GetCurrentDirectory())
+                      .AddJsonFile("appsettings.json");
+
+           var config = builder.Build();
+
+           var appConfig = new AppOptions();
+           config.GetSection("App").Bind(appConfig);
+
+           Console.WriteLine($"Height {appConfig.Window.Height}");
+       }
+   }
+
+   ````
+
+Using the following *appsettings.json* file:
+
+<!-- literal_block {"xml:space": "preserve", "language": "json", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "source": "/Users/shirhatti/src/Docs/aspnet/fundamentals/configuration/sample/src/ObjectGraph/appsettings.json", "highlight_args": {"linenostart": 1}, "names": []} -->
+
+````json
+
+   {
+     "App": {
+       "Profile": {
+         "Machine": "Rick"
+       },
+       "Connection": {
+         "Value": "connectionstring"
+       },
+       "Window": {
+         "Height": "11",
+         "Width": "11"
+       }
+     }
+   }
+   ````
+
+The program displays `Height 11`.
+
+The following code can be used to unit test the configuration:
+
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "highlight_args": {}, "names": []} -->
+
+````c#
+
+   [Fact]
+   public void CanBindObjectTree()
+   {
+       var dict = new Dictionary<string, string>
+               {
+                   {"App:Profile:Machine", "Rick"},
+                   {"App:Connection:Value", "connectionstring"},
+                   {"App:Window:Height", "11"},
+                   {"App:Window:Width", "11"}
+               };
+       var builder = new ConfigurationBuilder();
+       builder.AddInMemoryCollection(dict);
+       var config = builder.Build();
+
+       var options = new AppOptions();
+       config.GetSection("App").Bind(options);
+
+       Assert.Equal("Rick", options.Profile.Machine);
+       Assert.Equal(11, options.Window.Height);
+       Assert.Equal(11, options.Window.Width);
+       Assert.Equal("connectionstring", options.Connection.Value);
+   }
+   ````
 
 <a name=custom-config-providers></a>
 
-## Writing custom providers
+  ## Entity Framework custom provider
```

Comment:
Consider stressing that this is just a sample rather than best practices by e.g. adding the word "sample" in the title.